google-cloud-spanner/src/test/java/com/google/cloud/spanner/SpannerOptionsTest.java

```diff
@@ -918,4 +918,38 @@ public void testCustomAsyncExecutorProvider() {
             .build();
     assertSame(service, options.getAsyncExecutorProvider().getExecutor());
   }
+
+  @Test
+  public void testDefaultNumChannelsWithGrpcGcpExtensionEnabled() {
+    SpannerOptions options =
+        SpannerOptions.newBuilder().setProjectId("test-project").enableGrpcGcpExtension().build();
+
+    assertEquals(options.getNumChannels(), SpannerOptions.GRPC_GCP_ENABLED_DEFAULT_CHANNELS);
+  }
+
+  @Test
+  public void testNumChannelsWithGrpcGcpExtensionEnabled() {
+    // Set number of channels explicitly, before enabling gRPC-GCP channel pool in SpannerOptions
+    // builder.
+    int numChannels = 5;
```

Comment:
Would you mind adding a couple of tests to verify the behavior when `numChannels` is not set in combination with `GrpcGcpExension`? So verify that:
1. Gcp disabled: `numChannels == 4`
2. Gcp enabled: `numChannels == 8`

Also, can you verify that calling `SpannerOptions.toBuilder().build()` creates a correct new instance of `SpannerOptions` when Gcp has been enabled / disabled.

Reply:
@olavloite 
Can you please help me with these queries?
1. toBuilder() is a non-static method and cannot be referenced. Is there any other way to call SpannerOptions.toBuilder().build()?
2. What do you mean by correct new instance?

Reply:
@olavloite 
For verifying that a new instance is created, is the below test sufficient?

```
@Test
  public void checkCreatedInstanceWhenGrpcGcpEnabled() {
    SpannerOptions options =
        SpannerOptions.newBuilder()
            .setProjectId("test-project")
            .enableGrpcGcpExtension()
            .build();
    SpannerOptions options1 = options.toBuilder().build();

    Spanner spanner1 = options.getService();
    Spanner spanner2 = options1.getService();

    assertThat(spanner1 == spanner2, is(false));
  }
```

Reply:
(Sorry for the slow reply)

Yes, the above is what I meant, but it should also include something like `assertEquals(options, options1)` (or at least verify that both have the same setting for `enableGrpcGcpExtension`, as that is what we are adding here).

nit: We don't use `assertThat` anymore. Prefer `assertNotSame(spanner1, spanner2)` instead.

Reply:
@olavloite 
I tried comparing `options` and `options1` without enabling gRPC-GCP extension, but the comparison returns `false`. The code that I used to test is as follows:
```
SpannerOptions options =
        SpannerOptions.newBuilder()
            .setProjectId("test-project")
            .build();
SpannerOptions options1 = options.toBuilder().build();
System.out.println(options.equals(options1)); // prints false
```

Is there any other way we can verify that the setting is same?

Reply:
> at least verify that both have the same setting for `enableGrpcGcpExtension`

I have added a test in [commit](https://github.com/googleapis/java-spanner/pull/1997/commits/68e72d4cb33a52837461898c176f03dc650715bd#diff-d49ab467cce8baaefc131cef75c5713fc420f2c468d13a10ed56015be721715bR982-R983) that verifies `enableGrpcGcpExtension` has the same setting for both the options created.